### PR TITLE
[Merged by Bors] - perf(Ideal/Quotient): directly use AddCommGroup instance in CommRing

### PR DESCRIPTION
**Pull request by alreadydone**
with a bigger effect when applied on top of #17930

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alreadydone** [comment]
!bench

**Comment by grunweg** [comment]
Can you add a comment to the instance you modified --- so future readers don't try to simplify it again (and regress mathlib)?
Otherwise: this is a very small change, the new code remains readable (just gets one line longer) and yields significant gain, so seems worth taking.
maintainer delegate

**Comment by riccardobrasca** [comment]
Adding a comment seems a good idea, thanks!

bors d+

File: Mathlib/RingTheory/Ideal/Quotient/Defs.lean [hunk 1]
```diff
@@ -63,8 +63,9 @@ protected def ringCon (I : Ideal R) : RingCon R :=
         rw [mul_sub, sub_mul, sub_add_sub_cancel, mul_comm, mul_comm b₁]
       rwa [← this] at F }
 
-instance commRing (I : Ideal R) : CommRing (R ⧸ I) :=
-    inferInstanceAs (CommRing (Quotient.ringCon I).Quotient)
+instance commRing (I : Ideal R) : CommRing (R ⧸ I) where
```
Comment (alreadydone) [inline Mathlib/RingTheory/Ideal/Quotient/Defs.lean:66]:
```suggestion
-- This instance makes use of the existing AddCommGroup instance to boost performance.
instance commRing (I : Ideal R) : CommRing (R ⧸ I) where
```

Comment (alreadydone) [reply Mathlib/RingTheory/Ideal/Quotient/Defs.lean:66]: This modification to [switch to the `ring` instance](https://github.com/leanprover-community/mathlib4/blob/77fe4cd4369879746c0657d048678bc8090b6aef/Mathlib/RingTheory/Ideal/Quotient/Defs.lean#L52-L57) in #17930, but I think there's no harm getting it merged now.

**Review by alreadydone — COMMENTED** [review]
bors r+

**Pull request closed without merging**